google-cloud-examples/src/main/java/com/google/cloud/examples/bigtable/InstanceAdmin.java

```diff
@@ -0,0 +1,213 @@
+import com.google.cloud.bigtable.admin.v2.models.Instance;
+import com.google.cloud.bigtable.admin.v2.models.Instance.Type;
+import com.google.cloud.bigtable.admin.v2.models.StorageType;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class InstanceAdmin {
+
+  public static void main(String[] args) throws IOException {
+
+    String GCLOUD_PROJECT = System.getenv("GCLOUD_PROJECT");
+
+    if (GCLOUD_PROJECT.length() == 0) {
+      throw new Error("Environment variables GCLOUD_PROJECT must be set!");
+    }
+
+    BigtableInstanceAdminSettings instanceAdminSettings =
+        BigtableInstanceAdminSettings.newBuilder().setProjectName(ProjectName.of(GCLOUD_PROJECT))
+            .build();
+
+    BigtableInstanceAdminClient adminClient =
+        BigtableInstanceAdminClient.create(instanceAdminSettings);
+
+    System.out.println("Create an instance (type: PRODUCTION) and run basic instance-operations");
+    runInstanceOperations(adminClient, "ssd-instance", "ssd-cluster");
+
+    System.out.println("Create DEVELOPMENT instance");
+    createDevInstance(adminClient, "hdd-instance", "hdd-cluster");
+
+    System.out.println("Delete the Instance");
+    deleteInstance(adminClient, "hdd-instance");
+
+    System.out.println("Add Cluster");
+    addCluster(adminClient, "ssd-instance", "ssd-cluster");
+
+    System.out.println("Delete the Cluster");
+    deleteCluster(adminClient, "ssd-instance", "ssd-cluster");
+
+    // end operations with deleting the pro-instance created in `runInstanceOperations`
+    deleteInstance(adminClient, "ssd-instance");
+  }
+
+  public static void runInstanceOperations(BigtableInstanceAdminClient adminClient,
+      String instanceID, String clusterID) {
+    System.out.println("Check Instance Exists");
+    // [START bigtable_check_instance_exists]
+    boolean found = false;
+    try {
+      found = adminClient.exists(instanceID);
+    } catch (Exception e) {
+      System.out.println("Error checking if Instance exists: " + e.getMessage());
+    }
+
+    // Create instance if does not exists
+    if (!found) {
+      System.out.println("Creating a PRODUCTION Instance");
+      // [START bigtable_create_prod_instance]
+      // Creates a Production Instance with the ID "ssd-instance"
+      // with cluster id "ssd-cluster", 3 nodes and location us-central1-f
+      CreateInstanceRequest createInstanceRequest = CreateInstanceRequest.of(instanceID)
+          .addCluster(clusterID, "us-central1-f", 3, StorageType.SSD).setType(Type.PRODUCTION)
+          .addLabel("prod-label", "prod-label");
+      // Create production instance with given request
+      try {
+        Instance instance = adminClient.createInstance(createInstanceRequest);
+        System.out
+            .println("PRODUCTION type instance : " + instance.getId() + " created successfully");
+
+      } catch (Exception e) {
+        System.out.println("Error creating prod-instance: " + e.getMessage());
```

Comment:
If it fails to create an instance, I don't think we should just continue. It will break the rest of the operations with not_found errors